server/apps/publish/formatters/aap_ipnews_formatter.py

```diff
@@ -122,3 +174,110 @@ def format(self, article, subscriber):
 
     def can_format(self, format_type, article):
         return format_type == 'AAP IPNEWS' and article['type'] in ['text', 'preformatted']
+
+    def _set_selector_codes(self, article, subscriber_name, odbc_item, category):
+        desk_id = article.get('task', {}).get('desk', None)
+        if desk_id:
+            desk = superdesk.get_resource_service('desks').find_one(req=None, _id=desk_id)
+            if desk['name'] == 'NDX':
+                self._set_ndx_selector_codes(article, subscriber_name, odbc_item, category)
+            elif desk['name'] == 'SDX':
+                self._set_sdx_selector_codes(article, subscriber_name, odbc_item, category)
+            elif desk['name'] == 'FDX':
+                self._set_fdx_selector_codes(article, subscriber_name, odbc_item, category)
+
+    def _set_ndx_selector_codes(self, article, subscriber_name, odbc_item, category):
+        if 'MONITOR' in article['slugline'].upper():
+            odbc_item['selector_codes'] = self.SELECTOR_CODES['monitor'][subscriber_name]
+            return
+        if 'DIARY' in article['slugline'].upper():
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name, 'printsubs', 'cnewsd', 'rnewsd')
+            return
+        if 'FRONTERS' in article['slugline'].upper():
```

Comment:
It kind of makes sense to extract `article['slugline'].upper()` into a variable instead of repeatedly calling `.upper()` on the same thing.